scipy/stats/mstats_basic.py

```diff
@@ -356,39 +356,171 @@ def msign(x):
 
 
 def pearsonr(x, y):
-    """
-    Calculates a Pearson correlation coefficient and the p-value for testing
-    non-correlation.
+    r"""
+    Pearson correlation coefficient and p-value for testing non-correlation.
 
-    The Pearson correlation coefficient measures the linear relationship
-    between two datasets. Strictly speaking, Pearson's correlation requires
-    that each dataset be normally distributed. Like other correlation
+    The Pearson correlation coefficient [1]_ measures the linear relationship
+    between two datasets.  The calculation of the p-value relies on the
+    assumption that each dataset is normally distributed.  (See Kowalski [3]_
+    for a discussion of the effects of non-normality of the input on the
+    distribution of the correlation coefficient.)  Like other correlation
```

Comment:
Consider moving the part about p-value down to the next paragraph. It would fit perfectly there, but seems to interrupt this paragraph (in comparison).

Also, "relies on the assumption that each dataset is normally distributed": this is part of the null hypothesis and how one would derive the null distribution of the test statistic, but is that exactly what we mean to say about "each dataset"?

Reply:
agree

Reply:
How about the "dataset is normally distributed" part?

Do we mean that the _error_ "e" in your "Y = a + b*X + e" below is normal? I don't know if "each dataset is normally distributed" says that.

Reply:
I thought the description was ok: to calculate the p-value, it is assumed that x and y are samples of an underlying normal distribution and there is also a reference that explains how non-normality affects the p-value. in my example, if X and e are normal and independent, the weighted sum is also normal. I was not sure if any further clarification is needed

Reply:
I think when you wrote that "(x, y) is _jointly_ normal" it was more clear to me. "each dataset is normally distributed" suggests  to me that `x` and `y` are independently sampled from (possibly distinct) univariate normal distributions.